app/controllers/markets_controller.rb

```diff
@@ -0,0 +1,36 @@
+class MarketsController < ApplicationController
+
+  before_action only:[:show, :edit] { @market = Market.find(params[:id]) }
```

Comment:
Nice use of the before action! This is a lot in one line though, so I may extract the `find` logic out into a separate method